### PR TITLE
Benchmarking Script for both Relevance and KPI Inference Models

### PR DESCRIPTION
**Pull request by rishirich**
Signed-off-by: [Rishikesh Gawade](https://github.com/rishirich/)

Related to Issue/Enchancement  #184 

Contains modified inferencing coded for both Relevance and KPI Inferencing phases of the inference pipeline to add benchmarking steps.
Both models have been benchmarked thoroughly and includes the following metrics:
1) Relevance Model:
    - Total Number of Data Points Processed
    - Total Inference Time
    - Average Number of Pages Per PDF
    - Average Inference Time Per PDF
    - Minimum Inference Time of PDF
    - Maximum Inference Time of PDF
    - Std of Inference Times of PDFs
    - Average Time Per Data Point Processed
    - Average Data Points Processed Per Second
2) KPI Model:
    - Total Number of Data Points Processed
    - Total Inference Time 
    - Average Inference Time Per CSV
    - Minimum Inference Time of CSV 
    - Maximum Inference Time of CSV
    - Std of Inference Times of CSVs
    - Average Time Per Data Point Processed
    - Average Data Points Processed Per Second

### CODE REVIEW

File: notebooks/demo2/benchmarks.ipynb [hunk 1]
```diff
@@ -0,0 +1,1612 @@
+{
```
Comment (Shreyanand) [inline notebooks/demo2/benchmarks.ipynb:1]: <p>Add a title markdown explaining what this notebook is doing</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/os-climate/aicoe-osc-demo/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (Shreyanand) [inline notebooks/demo2/benchmarks.ipynb:1]: <p>Remove this cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/os-climate/aicoe-osc-demo/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (Shreyanand) [inline notebooks/demo2/benchmarks.ipynb:1]: <p>Remove this cell</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/os-climate/aicoe-osc-demo/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (Shreyanand) [inline notebooks/demo2/benchmarks.ipynb:1]: <p>Print this in a readable format and add a conclusion with results that you obtained.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/os-climate/aicoe-osc-demo/pull/194/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>

**Review by Shreyanand — COMMENTED** [review]
Good work :+1: Added some comments to fix.

Do the following in this a terminal window:

pip install pre-commit 

pre-commit run --all-files (cd into the repo first)

This will fix pre-commit errors.

Add the changes into a new commit.

Do a _git rebase -i HEAD~3_ (to squash the 3 commits you made into 1 commit)

Then update the common commit message and sign it off like in other PRs to satify DCO check.
that means end the commit message with  _Signed-off-by: your name <your id>_

**Comment by rishirich** [comment]
<p>Signed-off by: <a href="https://github.com/rishirich/" target="_blank">Rishikesh Gawade</a></p><p>Made the necessary changes, added comments and conclusion to the notebook.</p>

**Review by Shreyanand — APPROVED** [review]
The notebook contains script for benchmarking inference time for relevance and kpi-extraction tasks. The remaining things to do are: a) run it for 145 files b) get model performance metrics as well.
/lgtm

**Pull request merged**